### PR TITLE
Feature/registration-api // Add Registration API

### PR DESCRIPTION
**Pull request by Aslanyan99n**
- Add Registration API
- Add navigation to signIn screen after registration

### CODE REVIEW

File: Birthday/Birthday/Modules/Auth/Registration/Model/User.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+struct User {
+  
+  let id: Int?
+  let image: String?
+  let firstName: String?
+  let lastName: String?
+  let email: String?
+  
+  init(id: Int?, image: String?, firstName: String?, lastName: String?, email: String?) {
+    self.id = id
```
Comment (Mekhak11) [inline Birthday/Birthday/Modules/Auth/Registration/Model/User.swift:26]: multiline will be better.

Comment (Aslanyan99n) [reply Birthday/Birthday/Modules/Auth/Registration/Model/User.swift:26]: Changed

File: Birthday/Birthday/GraphQL/Repository/RegistrationRepository.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import BirthDayAPI
+
+protocol RegistrationRepository: GraphQLRepository {
+  
+  func singUp(
+    firstName: String,
+    lastName: String,
+    email: String,
+    password: String
+  ) -> AnyPublisher<SignUpMutation.Data, Error>
```
Comment (Mekhak11) [inline Birthday/Birthday/GraphQL/Repository/RegistrationRepository.swift]: It is better to have payload in this case, like struct RegistrationPayload  and as parameter give that one.

Comment (Aslanyan99n) [reply Birthday/Birthday/GraphQL/Repository/RegistrationRepository.swift]: Changed

**Review by Mekhak11 — APPROVED** [review]

**Review by nellyAslanyan — APPROVED** [review]

**Pull request merged**